p2p/simulations/adapters/inproc.go

```diff
@@ -161,39 +154,60 @@ func (self *SimNode) ServeRPC(conn net.Conn) error {
 	return nil
 }
 
-// Start starts the RPC handler and the underlying service
-func (self *SimNode) Start() error {
+// Start initializes the service, starts the RPC handler and then starts
+// the service
+func (self *SimNode) Start(snapshot []byte) error {
+	service := self.serviceFunc(self.Id, snapshot)
+
+	// for simplicity, only support single protocol services (simulating
+	// multiple protocols on the same peer is extra effort, and we don't
+	// currently run any simulations which run multiple protocols)
+	if len(service.Protocols()) != 1 {
+		return errors.New("service must have a single protocol")
+	}
+
 	self.dropPeers = make(chan struct{})
-	if err := self.startRPC(); err != nil {
+	if err := self.startRPC(service); err != nil {
 		return err
 	}
-	return self.service.Start(self)
+	self.running = service
+	return service.Start(&simServer{self})
+}
+
+// simServer wraps a SimNode but modifies the Start method signature so that
+// it implements the p2p.Server interface (the Start method is never actually
+// called when using the SimAdapter)
+type simServer struct {
```

Comment:
The sim node needs to implement both the `adapters.Node` interface and the `p2p.Server` interface, but they have conflicting `Start` method signatures, so we wrap `SimNode` in `simServer` to implement `p2p.Server.Start`, leaving `SimNode` implementing `adapters.Node.Start`.

Makes sense?